bot/utils/services.py

```diff
@@ -9,18 +7,37 @@
 log = get_logger(__name__)
 
 FAILED_REQUEST_ATTEMPTS = 3
+MAX_PASTE_LENGTH = 100_000
+
+
+class PasteUploadError(Exception):
+    """Raised when an error is encountered uploading to the paste service."""
+
+
+class PasteTooLongError(Exception):
+    """Raised when content is too large to upload to the paste service."""
 
 
-async def send_to_paste_service(contents: str, *, extension: str = "") -> Optional[str]:
+async def send_to_paste_service(contents: str, *, extension: str = "", max_length: int = MAX_PASTE_LENGTH) -> str:
     """
     Upload `contents` to the paste service.
 
-    `extension` is added to the output URL
+    `extension` is added to the output URL. `max_length` can be used to limit the allowed contents length
+    to lower than the maximum allowed by the paste service.
 
-    When an error occurs, `None` is returned, otherwise the generated URL with the suffix.
+    Raises `PasteTooLongError` if contents is too long to upload, and `PasteUploadError` if uploading fails.
+
+    Returns the generated URL with the extension.
     """
     extension = extension and f".{extension}"
-    log.debug(f"Sending contents of size {len(contents.encode())} bytes to paste service.")
+
+    max_size = min(max_length, MAX_PASTE_LENGTH)
```

Comment:
Do you think it would be better to raise `ValueError` if a length larger than `MAX_PASTE_LENGTH` is specified?

Reply:
If that goes uncaught, won't it go to Sentry? Consider using the custom exceptions again?

Reply:
Well, arguably it should go to Sentry since it's an error by the programmer to try to specify a limit higher than supported. Currently, it silently clamps it to the limit, meaning if the programmer actually intended for the limit to be higher, they wouldn't find out that it's impossible.

Reply:
Ah, I see now, I read this incorrectly as the code that somehow minified the input. Yes I agree, specifying a maximum limit higher than what the paste service actually supports should not fail silently.

Reply:
Yeah that makes sense 👍